diff --git a/grimoire_elk/enriched/github.py b/grimoire_elk/enriched/github.py
index 5aae3bce..f28da00e 100644
--- a/grimoire_elk/enriched/github.py
+++ b/grimoire_elk/enriched/github.py
@@ -493,6 +493,27 @@ class GitHubEnrich(Enrich):
                 'assignee_geolocation': assignee.get('geolocation', None)
             })
 
+        rich_pr['reviewer_data'] = list()
+        reviews_data = pull_request.get('reviews_data', list())
+        for review in reviews_data:
+            user_login = review.get('user_data', {}).get('login')
+            if not user_login:
+               continue
+
+            rich_pr['reviewer_data'].append({
+                'review_author_association': review.get('author_association', None),
+                'review_comment': review.get('body', None),
+                'review_commit_id': review.get('commit_id', None),
+                'review_state': review.get('state', None),
+                'review_submitted_at': review.get('submitted_at', None),
+                'review_html_url': review.get('html_url', None),
+                'review_user_login': review.get('user_data', {}).get('login'),
+                'review_user_id': review.get('user_data', {}).get('id', None),
+                'review_user_avatar_url': review.get('user_data', {}).get('avatar_url', None),
+                'review_user_html_url': review.get('user_data', {}).get('html_url', None),
+                'review_user_url': review.get('user_data', {}).get('url', None),
+                'review_user_location': review.get('user_data', {}).get('location', None),
+                'review_user_name': review.get('user_data', {}).get('name', None),
+                'review_user_company': review.get('user_data', {}).get('company', None),
+                'review_user_organizations': [x['login'] for x in review.get('user_data', {}).get('organizations', [])],
+            })
+
         rich_pr['id'] = pull_request['id']
         rich_pr['id_in_repo'] = pull_request['html_url'].split("/")[-1]
         rich_pr['repository'] = self.get_project_repository(rich_pr)
diff --git a/grimoire_elk/enriched/github2.py b/grimoire_elk/enriched/github2.py
index 81eb756d..9e6381c2 100644
--- a/grimoire_elk/enriched/github2.py
+++ b/grimoire_elk/enriched/github2.py
@@ -496,6 +496,27 @@ class GitHubEnrich2(Enrich):
                 'assignee_geolocation': assignee.get('geolocation', None)
             })
 
+        rich_pr['reviewer_data'] = list()
+        reviews_data = pull_request.get('reviews_data', list())
+        for review in reviews_data:
+            rich_pr['reviewer_data'].append({
+                'review_author_association': review.get('author_association', None),
+                'review_comment': review.get('body', None),
+                'review_commit_id': review.get('commit_id', None),
+                'review_state': review.get('state', None),
+                'review_submitted_at': review.get('submitted_at', None),
+                'review_html_url': review.get('html_url', None),
+                'review_user_login': review.get('user_data', {}).get('login'),
+                'review_user_id': review.get('user_data', {}).get('id', None),
+                'review_user_avatar_url': review.get('user_data', {}).get('avatar_url', None),
+                'review_user_html_url': review.get('user_data', {}).get('html_url', None),
+                'review_user_url': review.get('user_data', {}).get('url', None),
+                'review_user_location': review.get('user_data', {}).get('location', None),
+                'review_user_name': review.get('user_data', {}).get('name', None),
+                'review_user_company': review.get('user_data', {}).get('company', None),
+                'review_user_organizations': [x['login'] for x in review.get('user_data', {}).get('organizations', [])],
+            })
+
         rich_pr['id'] = pull_request['id']
         rich_pr['pull_id'] = pull_request['id']
         rich_pr['pull_id_in_repo'] = pull_request['html_url'].split("/")[-1]
