diff --git a/grimoire_elk/enriched/github2.py b/grimoire_elk/enriched/github2.py
index 284b353..65c68a5 100644
--- a/grimoire_elk/enriched/github2.py
+++ b/grimoire_elk/enriched/github2.py
@@ -29,6 +29,7 @@ from .utils import get_time_diff_days
 
 from .enrich import Enrich, metadata
 from ..elastic_mapping import Mapping as BaseMapping
+from .identity import emails, get_token, get_email
 
 
 MAX_SIZE_BULK_ENRICHED_ITEMS = 200
@@ -42,7 +43,7 @@ REVIEW_COMMENT_TYPE = 'review_comment'
 REPOSITORY_TYPE = 'repository'
 
 logger = logging.getLogger(__name__)
-
+access_token = get_token()
 
 class Mapping(BaseMapping):
 
@@ -167,6 +168,14 @@ class GitHubEnrich2(Enrich):
         identity['username'] = user.get('username', None)
         identity['email'] = user.get('email', None)
 
+        if not identity['email']:
+            email = emails.get(identity['username'], None)
+            if not email:
+                email = get_email(access_token, identity['username'])
+            if email:
+                emails[identity['username']] = email
+                identity['email'] = email
+
         return identity
 
     def get_project_repository(self, eitem):
