diff --git a/grimoire_elk/enriched/jira.py b/grimoire_elk/enriched/jira.py
index b2093e9..781c2a7 100644
--- a/grimoire_elk/enriched/jira.py
+++ b/grimoire_elk/enriched/jira.py
@@ -28,7 +28,7 @@ from .enrich import Enrich, metadata, SH_UNKNOWN_VALUE
 from ..elastic_mapping import Mapping as BaseMapping
 
 from .utils import get_time_diff_days
-
+from .identity import emails, get_token, get_email
 
 MAX_SIZE_BULK_ENRICHED_ITEMS = 200
 ISSUE_TYPE = 'issue'
@@ -36,6 +36,7 @@ COMMENT_TYPE = 'comment'
 
 logger = logging.getLogger(__name__)
 
+access_token = get_token()
 
 class Mapping(BaseMapping):
 
@@ -104,6 +105,19 @@ class JiraEnrich(Enrich):
         if 'emailAddress' in user:
             identity['email'] = user['emailAddress']
 
+        if not identity['username']:
+            return identity
+
+        if not identity['email']:
+            email = emails.get(identity['username'], None)
+            if not email:
+                email = get_email(access_token, identity['username'])
+                if email:
+                    emails[identity['username']] = email
+                    identity['email'] = email
+            else:
+                identity['email'] = email
+
         return identity
 
     def get_item_sh(self, item, roles=None, date_field=None):
