diff --git a/perceval/backends/core/github.py b/perceval/backends/core/github.py
index f7ae3a4..628a756 100644
--- a/perceval/backends/core/github.py
+++ b/perceval/backends/core/github.py
@@ -71,7 +71,7 @@ DEFAULT_SLEEP_TIME = 1
 MAX_RETRIES = 5
 
 TARGET_ISSUE_FIELDS = ['user', 'assignee', 'assignees', 'comments', 'reactions']
-TARGET_PULL_FIELDS = ['user', 'review_comments', 'requested_reviewers', "merged_by", "commits"]
+TARGET_PULL_FIELDS = ['user', 'review_comments', 'requested_reviewers', 'merged_by', 'commits', 'assignee', 'assignees']
 
 logger = logging.getLogger(__name__)
 
@@ -362,6 +362,10 @@ class GitHub(Backend):
                     pull[field + '_data'] = self.__get_pull_requested_reviewers(pull['number'])
                 elif field == 'commits':
                     pull[field + '_data'] = self.__get_pull_commits(pull['number'])
+                elif field == 'assignee':
+                    pull[field + '_data'] = self.__get_issue_assignee(pull[field])
+                elif field == 'assignees':
+                    pull[field + '_data'] = self.__get_issue_assignees(pull[field])
 
             yield pull
 
